pyccel/ast/core.py

```diff
@@ -5177,15 +5177,56 @@ def bodies(self):
         return b
 
 
-class IfTernaryOperator(If):
+class IfTernaryOperator(Basic, PyccelAstNode):
+    """Represents a IfTernaryOperator statement in the code.
 
-    """class for the Ternery operator"""
+    Parameters
+    ----------
+    args :
+        args : type list
+        format : [condition , value_if_true, value_if_false]
+
+    Examples
+    --------
+    >>> from sympy import Symbol
+    >>> from pyccel.ast.core import Assign, IfTernaryOperator
+    >>> n = Symbol('n')
+    >>> x = 5 if n > 1 else 2
+    >>> IfTernaryOperator([PyccelGt(n > 1),  5,  2])
+    IfTernaryOperator([PyccelGt(n > 1),  5,  2])
+    """
     def __init__(self, *args):
-        for arg in self.args:
-            if len(arg[1].body)!=1:
-                raise TypeError('IfTernary body must be of length 1')
+        self._args = []
+        cond = args[0]
+        first = args[1]
+        second = args[2]
+        self._args = [cond, first, second]
+        if self.stage == 'syntactic':
+            return
 
-    pass
+        if isinstance(first , Nil) or isinstance(second, Nil):
+            raise TypeError('None is not implemeted for Ternary Operator')
```

Comment:
Could you use a `NotImplementedError` instead of a `TypeError`. We catch `NotImplementedError`s so they raise a neat error, but `TypeError`s should occur when a developer is using the code wrong as `TypeError`s will lead to back traces